src/Mutations/PostMutation.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace WPGraphQLPostsToPosts\Mutations;
+
+use WP_Post_Type;
+use WPGraphQLPostsToPosts\Types\Fields;
+
+class PostMutation extends AbstractMutation {
+	public function register_hooks() : void {
+		parent::register_hooks();
+		add_action( 'graphql_post_object_mutation_update_additional_data', [ $this, 'save_additional_data' ], 10, 4 );
+	}
+
+	public function register_input_fields() : void {
+		$post_types = self::get_post_types();
+
+		foreach ( $post_types as $post_type ) {
+			$graphql_single_name = $post_type->graphql_single_name;
+
+			register_graphql_field(
+				'Update' . $graphql_single_name . 'Input',
+				Fields::NAME,
+				[
+					'type'        => [ 'list_of' => Fields::MUTATION_TYPE ],
+					'description' => __( 'Id', 'wp-graphql-posts-to-posts' ),
+				]
+			);
+			register_graphql_field(
+				'Create' . $graphql_single_name . 'Input',
+				Fields::NAME,
+				[
+					'type'        => [ 'list_of' => Fields::QUERY_TYPE ],
+					'description' => __( 'Id', 'wp-graphql-posts-to-posts' ),
+				]
+			);
+		}
+	}
+
+	public function save_additional_data( int $post_id, array $input, WP_Post_Type $post_type_object, string $mutation_name ) : void {
+		$post_types = self::get_post_types();
+		foreach ( $post_types as $post_type ) {
+			if ( self::camel_case_to_underscores( $mutation_name ) !== $post_type->name ) {
+				continue;
+			}
+
+			$p2p_connections_to_map = Fields::get_p2p_connections();
+
+			$field_names = [];
+
+			$connection_name = $post_type->name;
+
+			$connections = array_filter( $p2p_connections_to_map, fn( $p2p_connection ) => $p2p_connection['from'] === $connection_name || $p2p_connection['to'] === $connection_name );
+
+			foreach ( $connections as $connection ) {
+				array_push( $field_names, $connection['name'] );
+			}
+
+			foreach ( $input['postToPostConnections'] as $post_to_post_connection ) {
+				if ( ! in_array( $post_to_post_connection['connection'], $field_names, true ) ) {
+					continue;
+				}
+
+				$connected_type = $post_to_post_connection['connection'];
+
+				if ( ( ! $post_to_post_connection['append'] || 0 === count( $post_to_post_connection['ids'] ) ) && false === strpos( $mutation_name, 'Create' ) ) {
+					$connection_to = null;
+
+					foreach ( $connections as $connection ) {
+						if ( $connection['name'] === $post_to_post_connection['connection'] ) {
+							$connection_to = $connection['to'];
```

Comment:
Should there be a `break` here?

Reply:
Not sure what you mean?

Reply:
It's a foreach loop for the sole purpose of setting a value under an if condition. Are we iterating through the entire array to set it on the last value, or break the foreach on the first match?